src/main/java/sg/ncl/MainController.java

```diff
@@ -2877,4 +2885,48 @@ private Realization getCleanRealization() {
 
         return realization;
     }
+
+    /**
+     * Computes the number of teams that the user is in and the number of running experiments to populate data for the user dashboard
+     * @return a map in the form teams: numberOfTeams, experiments: numberOfExperiments
+     */
+    private Map<String, Integer> getUserDashboardStats(String userId) {
+        int numberOfRunningExperiments = 0;
+        Map<String, Integer> userDashboardStats = new HashMap<>();
+
+        // get list of teamids
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity userRespEntity = restTemplate.exchange(properties.getUser(userId), HttpMethod.GET, request, String.class);
+
+        JSONObject object = new JSONObject(userRespEntity.getBody().toString());
+        JSONArray teamIdsJsonArray = object.getJSONArray("teams");
+
+        for (int i = 0; i < teamIdsJsonArray.length(); i++) {
+            String teamId = teamIdsJsonArray.get(i).toString();
+
+            HttpEntity<String> teamRequest = createHttpEntityHeaderOnly();
+            ResponseEntity teamResponse = restTemplate.exchange(properties.getTeamById(teamId), HttpMethod.GET, teamRequest, String.class);
+            String teamResponseBody = teamResponse.getBody().toString();
+
+            if (!isMemberJoinRequestPending(userId, teamResponseBody)) {
+                // get experiments lists of the teams
+                HttpEntity<String> expRequest = createHttpEntityHeaderOnly();
+                ResponseEntity expRespEntity = restTemplate.exchange(properties.getExpListByTeamId(teamId), HttpMethod.GET, expRequest, String.class);
+
+                JSONArray experimentsArray = new JSONArray(expRespEntity.getBody().toString());
+
+                for (int k = 0; k < experimentsArray.length(); k++) {
+                    Experiment2 experiment2 = extractExperiment(experimentsArray.getJSONObject(k).toString());
+                    Realization realization = invokeAndExtractRealization(experiment2.getTeamName(), experiment2.getId());
+                    if (realization.getState().equals(RealizationState.RUNNING.toString())) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)